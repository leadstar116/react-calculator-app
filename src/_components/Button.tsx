import React from 'react'

type Props = {
  name: string,
  style?: string,
  clickHandler: (buttonName: string) => void,
}

const Button = (props: Props) => {
  return (
    <div className={props.style ? "calculator-button " + props.style : "calculator-button"}>
      <button onClick={() => props.clickHandler(props.name)}>{props.name}</button>
    </div>
  );
}

export default Button