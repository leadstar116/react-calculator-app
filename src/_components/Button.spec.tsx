import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Button from './Button'

describe('Button Component', () => {

  let component = renderer.create(
    <BrowserRouter>
      <Button name="รท" clickHandler={(buttonName: string) => {}} style="orange" />
    </BrowserRouter>
  );

  it('should render with given props', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
})