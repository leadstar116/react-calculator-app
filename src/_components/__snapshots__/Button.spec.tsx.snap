// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render with given props 1`] = `
<div
  className="calculator-button orange"
>
  <button
    onClick={[Function]}
  >
    รท
  </button>
</div>
`;
